scripts/globals/job_utils/paladin.lua

```diff
@@ -0,0 +1,223 @@
+    if player:getShieldSize() == 0 then
+        return xi.msg.basic.REQUIRES_SHIELD, 0
+    else
+        ability:setRecast(ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST))
+
+        return 0, 0
+    end
+end
+
+xi.job_utils.paladin.checkInvincible = function(player, target, ability)
+    local jpValue = player:getJobPointLevel(xi.jp.INVINCIBLE_EFFECT)
+
+    ability:setVE(ability:getVE() + 100 * jpValue)
+    ability:setRecast(ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST))
+
+    return 0, 0
+end
+
+xi.job_utils.paladin.checkSepulcher = function(player, target, ability)
+    if target:isUndead() then
+        return 0, 0
+    else
+        return xi.msg.basic.CANNOT_ON_THAT_TARG, 0
+    end
+end
+
+xi.job_utils.paladin.checkShieldBash = function(player, target, ability)
+    if player:getShieldSize() == 0 then
+        return xi.msg.basic.REQUIRES_SHIELD, 0
+    else
+        return 0, 0
+    end
+end
+
+-----------------------------------
+-- Ability Use Functions
+-----------------------------------
+xi.job_utils.paladin.useChivalry = function(player, target, ability)
+    local merits = player:getMerit(xi.merit.CHIVALRY) - 5
+    local tp     = target:getTP()
+    local base   = 0.05 + (player:getMod(xi.mod.ENHANCES_CHIVALRY) / 100)
+    -- MP gained = (TP * 0.05) + (0.0015 * TP * MND) * Merits
+    local amount = (tp * base) + (0.0015 * tp * target:getStat(xi.mod.MND)) * ((100 + merits) / 100)
+
+    target:setTP(0)
+
+    return target:addMP(amount)
+end
+
+xi.job_utils.paladin.useCover = function(player, target, ability)
+    local baseDuration = 15
+    local bonusTime    = utils.clamp(math.floor((player:getStat(xi.mod.VIT) + player:getStat(xi.mod.MND) - target:getStat(xi.mod.VIT) * 2) / 4), 0, 15)
+    local jpValue      = player:getJobPointLevel(xi.jp.COVER_DURATION)
+    local duration     = baseDuration + bonusTime + player:getMerit(xi.merit.COVER_EFFECT_LENGTH) + player:getMod(xi.mod.COVER_DURATION) + jpValue
+
+    player:addStatusEffect(xi.effect.COVER, player:getMod(xi.mod.COVER_TO_MP), 0, duration)
+    player:setLocalVar("COVER_ABILITY_TARGET", target:getID())
+    ability:setMsg(xi.msg.basic.COVER_SUCCESS)
+end
+
+xi.job_utils.paladin.useDivineEmblem = function(player, target, ability)
+    -- Divine Magic bonus damage handled in globals/magic.lua
+    local power = 50 + player:getMod(xi.mod.ENHANCES_DIVINE_EMBLEM) -- 50% increase to enmity
+
+    player:addStatusEffect(xi.effect.DIVINE_EMBLEM, power, 0, 60)
+end
+
+xi.job_utils.paladin.useFealty = function(player, target, ability)
+    local merits    = player:getMerit(xi.merit.FEALTY) - 5
+    local enhFealty = (player:getMerit(xi.merit.FEALTY) / 5) * player:getMod(xi.mod.ENHANCES_FEALTY)
+    local duration  = 60 + merits + enhFealty
+
+    player:addStatusEffect(xi.effect.FEALTY, 1, 0, duration)
+end
+
+xi.job_utils.paladin.useHolyCircle = function(player, target, ability)
+    -- TODO: Create xi.mod.UNDEAD_SDT mod and core handling?
```

Comment:
probably don't need undead SDT, just need to check undead family vs holy circle effect power

Reply:
though on second thought a mod is much faster to lookup in theory... out of scope of this PR

Reply:
My main thought with making them mods was the future addition of the other Circle effect SDT’s and killer effect equivalent SDT’s. It would need to be adjustments in the core handling for damage that are pulled from the player (or jug/charmed pet) and would increase damage against that family.

Thinking about it further, calling them SDT’s may be a misnomer. They do affect a specific damage category though.

I haven’t gone back through and investigated how to accomplish the set up yet, definitely out of the scope of this PR. I can set up an issue with the TODO when the PR is merged.

Reply:
> Thinking about it further, calling them SDT’s may be a misnomer. They do affect a specific damage category though.

the language is def a problem, because sdt means a very different thing in our core.

Reply:
can we change the comment here to just reflect the unimplemented part with info from wiki without a suggested modifier name? like
```lua
-- 15% damage bonus against 15% damage resistance from undead, 5% when subbed
```
It's really a kind of damage _given_ mod combined with a not-specific damage taken

Reply:
Addressed in #2847.